storybookjs/storybook
# Enables Yarn 2

## What I did

Upgrading the Yarn binary to the latest stable release (2.4), which should make them faster.

I've configured the project to stay on the node_modules linker since it makes the upgrade fairly painless. Tests pass locally, I'll see how it goes on CI and tweak the command line references accordingly (things like `--frozen-lockfile` being renamed to `--immutable`, etc).

Notable changes:

- The Yarn binary is checked within this repository. This mean that all your contributors will now be ensured to use the exact same version of Yarn, regardless what they have installed globally.

- The root package.json now lists all workspaces as dependencies (using the `workspace:` protocol). This is because in Yarn 2 the root workspace doesn't implicitly depend on all workspaces anymore, they have to be explicitly listed. In theory the dependencies should be listed in each workspace that depends on them (so for example `ui` should depend on `api`) but I'm worried it would lead to a gigantic diff, so for now the repo keeps relying on the hoisting to work.

## How to test

The existing CI pipeline should be enough to validate that the upgrade is safe. One potential risky part is the publish workflow; I've heard it's already covered by tests using Verdaccio so it should be fine, but if someone can double check it would be perfect!
